### ISSUE TITLE
dropdown not showing whole respose on entering the keys with spaces

### ISSUE THREAD

**Issue by omprakash9** [Gbuomprisco/ngx-chips#319]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report 
[ ] feature request
[ ] support request
```

**Current behavior**
Issue 1:
If I enter keys with spaces, auto dropdown is not showing the entire returned object as expected . I think the reason is that the text entered with the spaces are treating it as one key(ex: sun flower). but I need it to treat it  as two different keys and dropdown should populate response from key of word "sun" and response from key of word "flower". can you please help me out on fixing this issue.

Issue 2:
I enter the text in search box, on hitting the Enter key, Text is getting cleared. Can you please suggest me on how to prevent the text not to get cleared on hitting the enter key. because I need the text to be displayed even after I get the results for the search I made through the text.

* **Angular version:** 2.4.5

* **ng2-tag-input version:**
1.0.0

* **Browser:** IE 11

**Comment by omprakash9** [comment]
Hi @Gbuomprisco ,if you don't mine,  can you please update the time frame for the above fix?
since our project needs to meet deadline

**Comment by Gbuomprisco** [comment]
Hi @omprakash9,

early next week or weekend

**Comment by omprakash9** [comment]
@Gbuomprisco , thanks and please let me know once it is fixed!

**Comment by Gbuomprisco** [comment]
Sorry I really don't understand point 1, can you rephrase?

**Comment by Gbuomprisco** [comment]
@omprakash9 have you tried using a custom matching function as described in the documentation? the way it works now is fairly standard - therefore I am not considering changing it for now, but that's why I put the custom matching function

**Comment by omprakash9** [comment]
OK I will try it, How about the second issue? did you have any suggestions for it?

**Comment by omprakash9** [comment]
@Gbuomprisco 
Can you please provide us a code example that implements the matchingFn input property.Because, i am wondering whether i have to remove autocompleteObservable property.

What I have done is I have added [matchingFn]= customFilter(value, requestAutocompleteItems)" in the tag-input-dropdown and calling customFilter in my local component by passing user searched text and ressponse from observable as second parameter.
Can you please correct me if it's wrong.

Any help would be highly appreciated..!

**Comment by omprakash9** [comment]
HI @Gbuomprisco 
 can you please provide me an example!

**Comment by Gbuomprisco** [comment]
@omprakash9 have a look at this comment as example https://github.com/Gbuomprisco/ng2-tag-input/issues/188#issuecomment-279477395

**Comment by omprakash9** [comment]
@Gbuomprisco , matchingFn filter is working. Thankyou very much

I'm curious about the issue2 can you please help me with a  suggestion!

**Comment by Gbuomprisco** [comment]
@omprakash9 what if you use the [inputText] attribute to reset the value on blur event? you'd need to store it somewhere

**Comment by ranouf** [comment]
Like @omprakash9 i m interesting to know how to use this function? 
Is there someone who can share his solution?

Thanks

**Comment by Gbuomprisco** [comment]
Hi @ranouf,

There is pretty much a whole implementation here https://github.com/Gbuomprisco/ng2-tag-input/issues/188#issuecomment-279477395. Define it in your parent component and pass it down to tag-input-dropdown. 

I see you defined `filter`, but you shouldn't add parenthesis to it. 

What exactly is not clear?
